plan/exhaust_physical_plans.go

```diff
@@ -185,30 +185,28 @@ func (p *LogicalJoin) getHashJoin(prop *requiredProp, innerIdx int) *PhysicalHas
 // joinKeysMatchIndex checks if all keys match columns in index.
 // It returns a slice a[] what a[i] means keys[i] is related with indexCols[a[i]].
 func joinKeysMatchIndex(keys, indexCols []*expression.Column, colLengths []int) []int {
-	if len(indexCols) < len(keys) {
-		return nil
-	}
 	keyOff2IdxOff := make([]int, len(keys))
-	for keyOff, key := range keys {
-		idxOff := joinKeyMatchIndexCol(key, indexCols, colLengths)
-		if idxOff == -1 {
-			return nil
-		}
-		keyOff2IdxOff[keyOff] = idxOff
-	}
-	return keyOff2IdxOff
-}
-
-func joinKeyMatchIndexCol(key *expression.Column, indexCols []*expression.Column, colLengths []int) int {
-	for idxOff, idxCol := range indexCols {
-		if colLengths[idxOff] != types.UnspecifiedLength {
+	for i := range keyOff2IdxOff {
+		keyOff2IdxOff[i] = -1
+	}
+	// There should be at least one column in join keys which can match the index's column.
+	matched := false
+	tmpSchema := expression.NewSchema(keys...)
+	for i, idxCol := range indexCols {
+		if colLengths[i] != types.UnspecifiedLength {
```

Comment:
can we use `idxCol.GetType().Flen` instead of `colLengths[i]`?

Reply:
No, it's the length of the index's column, not the actual column length.
`create table t(a varchar(50), index idx(a(10));`
The `flen` of `a` is 50 right?

Reply:
ok